packages/devtools_app/lib/src/screens/memory/panes/allocation_tracing/allocation_profile_tracing_tree.dart

```diff
@@ -159,8 +227,12 @@ class _AllocationProfileTracingTreeHeader extends StatelessWidget {
         CollapseAllButton(
           onPressed: () => updateTreeStateCallback(
             () {
+              final isBottomUp = tabs[tabController.index] ==
+                  AllocationTracingTree._bottomUpTab;
               final data = controller.selectedTracedClassAllocationData!;
-              for (final root in data.bottomUpRoots) {
+              final roots =
```

Comment:
create a helper _currentDataRoots that can be used both the expand and collapse button onPressed functions

Reply:
Done.